angularjs/angular.d.ts

```diff
@@ -651,6 +651,12 @@ declare namespace angular {
         [name: string]: any;
     }
 
+    interface IFilterOrderByObject {
+        value: any;
+        type: any;
+        index: any;
+    }
```

Comment:
We could probably type the `predicates` parameter as `(string | ((value: T) => U))[] | (value: T) => U | string` if type arguments could be inferred to be union types in the cases like `[x => 1, x => 'a']`. But unfortunately the inference [doesn't work like this](http://stackoverflow.com/questions/39905523/why-isnt-the-type-argument-inferred-as-a-union-type).